GVFS/GVFS.UnitTests/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,137 @@
+                ServiceDataLocation,
+                this.gvfsPlatformMock.Object,
+                mountProcessMock.Object);
+
+            repoRegistry.AutoMountRepos(int.Parse(ExpectedActiveUserId));
+
+            mountProcessMock.Verify(
+                mp => mp.MountRepository(
+                    It.Is<string>(repo => repo.Equals(ExpectedActiveRepoPath)),
+                    It.Is<int>(id => id == int.Parse(ExpectedActiveUserId)),
+                    It.IsAny<ITracer>()),
+                Times.Once,
+                $"{nameof(mountProcessMock.Object.MountRepository)} was not called for registered repository");
+        }
+
+        [TestCase]
+        public void MountProcessLaunchedUsingCorrectArgs()
+        {
+            string executable = @"/bin/launchctl";
+            string expectedArgs = $"asuser {int.Parse(ExpectedActiveUserId)} /usr/local/vfsforgit/gvfs mount {ExpectedActiveRepoPath}";
+
+            Mock<GVFSMountProcess.MountLauncher> mountLauncherMock = new Mock<GVFSMountProcess.MountLauncher>();
+            mountLauncherMock.Setup(mp => mp.LaunchProcess(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<ITracer>()))
+                .Returns(true);
+
+            GVFSMountProcess mountProcess = new GVFSMountProcess(mountLauncherMock.Object, waitTillMounted: false);
+            mountProcess.MountRepository(ExpectedActiveRepoPath, int.Parse(ExpectedActiveUserId), this.tracer);
+
+            mountLauncherMock.Verify(
+                mp => mp.LaunchProcess(
+                    It.Is<string>(exe => exe.Equals(executable)),
+                    It.Is<string>(args => args.Equals(expectedArgs)),
+                    It.Is<string>(workingDirectory => workingDirectory.Equals(ExpectedActiveRepoPath)),
+                    It.IsAny<ITracer>()),
+                Times.Once,
+                $"{nameof(mountLauncherMock.Object.LaunchProcess)} was not called for registered repository");
+        }
+
+        private void CreateTestRepos(PhysicalFileSystem fileSystem, string dataLocation)
+        {
+            string repo1 = Path.Combine("mock:", "code", "repo1");
+            string repo2 = ExpectedActiveRepoPath;
+            string repo3 = Path.Combine("mock:", "code", "repo3");
+            string repo4 = Path.Combine("mock:", "code", "repo4");
+
+            this.fileSystem.WriteAllText(
+                Path.Combine(dataLocation, RepoRegistry.RegistryName),
+                $@"1
+                {{""EnlistmentRoot"":""{repo1.Replace("\\", "\\\\")}"",""OwnerSID"":502,""IsActive"":false}}
+                {{""EnlistmentRoot"":""{repo2.Replace("\\", "\\\\")}"",""OwnerSID"":502,""IsActive"":true}}
+                {{""EnlistmentRoot"":""{repo3.Replace("\\", "\\\\")}"",""OwnerSID"":501,""IsActive"":false}}
+                {{""EnlistmentRoot"":""{repo4.Replace("\\", "\\\\")}"",""OwnerSID"":501,""IsActive"":true}}
+                ");

```

Comment:
I don't think you need to do thereplace.  The `Path.Combine` would have added the right delimiter.  There are 2 in the code before because it needed to be escaped.  Using `Path.Combine` it doesn't need to be.